src/Azure.Functions.Cli/Helpers/TelemetryHelpers.cs

```diff
@@ -0,0 +1,93 @@
+﻿using Azure.Functions.Cli.Common;
+using Azure.Functions.Cli.Telemetry;
+using Colors.Net;
+using Fclp.Internals;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading;
+
+namespace Azure.Functions.Cli.Helpers
+{
+    internal static class TelemetryHelpers
+    {
+        public static IEnumerable<string> GetCommandsFromCommandLineOptions(IEnumerable<ICommandLineOption> options)
+        {
+            return options.Select(option => option.HasLongName ? option.LongName : option.ShortName);
+        }
+
+        public static void AddCommandEventToDictionary(IDictionary<string, string> events, string eventName, string eventVal)
+        {
+            if (StaticSettings.IsTelemetryEnabled)
+            {
+                try
+                {
+                    events[eventName] = eventVal;
+                }
+                catch (Exception ex)
+                {
+                    if (StaticSettings.IsDebug)
+                    {
+                        ColoredConsole.Error.WriteLine(ex.ToString());
+                    }
+                }
+            }
+        }
+
+        public static void LogEventIfAllowedSafe(ITelemetry telemetry, TelemetryEvent telemetryEvent)
+        {
+            try
+            {
+                LogEventIfAllowed(telemetry, telemetryEvent);
+            }
+            catch
+            { 
+                // oh well!
+            }
+        }
+
+        public static void LogEventIfAllowed(ITelemetry telemetry, TelemetryEvent telemetryEvent)
```

Comment:
Wondering if this return a  boolean to indicate whether the event was logged.

Reply:
I can make that. But right now, I don't see the need to use the boolean value. We could have a debug log for whether or not we logged it I suppose. But I figured, if Telemetry is off, I would make it completely off (no logs about it etc).
But I also don't see a strong reason to not log that "telemetry logging was not allowed".

Reply:
Not super important... just something I noticed and wanted to bring up.